api/dps/server.go

```diff
@@ -31,28 +32,29 @@ import (
 // This is generally an on-disk interface, but could be a GRPC-based index as
 // well, in which case there is a double redirection.
 type Server struct {
-	index dps.Reader
-	codec dps.Codec
-
+	index    dps.Reader
+	codec    dps.Codec
+	tracer   trace.Tracer
 	validate *validator.Validate
 }
 
 // NewServer creates a new server, using the provided index reader as a backend
 // for data retrieval.
-func NewServer(index dps.Reader, codec dps.Codec) *Server {
+func NewServer(index dps.Reader, codec dps.Codec, tracer trace.Tracer) *Server {
```

Comment:
It might be better to have a default `trace.NewNoopTracer()` but allow NewServer to accept options where you could set a custom tracer, since the tracer is not mandatory for the server to function.

Reply:
Done, Made a server config so it's extensible